addons/mass_mailing/models/mail_mail.py

```diff
@@ -104,3 +104,11 @@ def _postprocess_sent_message(self, success_pids, failure_reason=False, failure_
         else:
             self.filtered('mailing_id').mailing_trace_ids.set_sent()
         return super()._postprocess_sent_message(success_pids, failure_reason=failure_reason, failure_type=failure_type)
+
+    @api.autovacuum
+    def _gc_canceled_mail_mail(self):
+        """This method garbage collects canceled mail.mail records."""
+        # The 10000 limit is arbitrary, chosen a big limit so that the cleaning can be shorter and not too big so that we don't block the server
+        canceled_mails = self.with_context(active_test=False).env['mail.mail'].search([('state', '=', 'cancel')], limit=10000)
```

Comment:
Can probably begin with older one, + useless env
```suggestion
        canceled_mails = self.with_context(active_test=False).search([('state', '=', 'cancel')], order='id sac', limit=10000)
```

Reply:
We could use an optional config parameter, if unset, 6 months is probably good enough.